static/js/vue-cdr-access/src/components/full_record/fileList.vue

```diff
@@ -240,9 +247,79 @@ export default {
             return { markDeleted: markedForDeletion, restricted: restrictedAccess };
         },
 
-        ariaLabelText(row) {
-            return this.$t('full_record.edit_title', { title: row.title });
+        showDropdownList(e) {
+            // Close any currently open dropdowns
+            this.closeDropdownLists(e);
+
+            if (e.target.id.startsWith('dcr-download')) {
+                let drop_down = e.target.parentElement.parentElement.querySelector('.dropdown-menu');
+                if (drop_down !== null) {
+                    drop_down.setAttribute('aria-hidden', 'false');
+                    drop_down.classList.add('show-list');
+                }
+            }
+        },
+
+        closeDropdownLists() {
+            document.querySelectorAll('.show-list').forEach(element => {
+                element.setAttribute('aria-hidden', 'true');
+                element.classList.remove('show-list');
+            });
+        },
+
+        downloadButtonHtml() {
```

Comment:
I am guessing this had to be duplicated because the html of the datatable is dynamically generated, so there wasn't a place to add in the fileDownload.vue component? It looks like we do have access to the current vue component from inside of the datatable though, so we may be able to cut down on duplication.

One option would be to replace fileDownload.vue with the fileDownloadUtils mixin, using this method (receiving the brief_object as a param) to generate the html instead of a template everywhere. Then I think we could call downloadButtonHtml where ever we need that element. Preferably the mixin could include the dropdown closing logic too.

It might also be possible to dynamically add another component via generated html being added to the page, either by some programmatic means of getting vue to import a component at a particular place in the html, or maybe even just including the `<fileDownload></fileDownload>` tag in the generated HTML if its going to be added to the page inside of the area controlled by vue. My impression is that vuejs is supposed to evaluate insertions to the HTML in these scenarios, but I'm not totally sure.

Reply:
What about using `downloadButtonHtml` instead of fileDownload.vue everywhere? So that we would just have one implementation of it.